### PR TITLE
IGNITE-17278 TableManager#directTableIds optimized

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-17278

### CODE REVIEW

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/NamedListConfiguration.java [hunk 1]
```diff
@@ -122,6 +123,15 @@ public T getByInternalId(UUID internalId) {
         return (T) members.get(value.keyByInternalId(internalId));
     }
 
+    /**
+     * Returns all internal ids of the elements from the list.
+     */
+    public List<UUID> internalIds() {
+        NamedListView<VIEWT> value = value();
+
+        return new ArrayList<>(((NamedListNode) value).internalIds());
```
Comment (tkalkirill) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/NamedListConfiguration.java:132]: How about an immutable list?

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java [hunk 2]
```diff
@@ -834,6 +834,10 @@ public T visitNamedListNode(String key, NamedListNode<?> node) {
 
                         assert pathNode.namedListEntry;
 
+                        if (!pathNode.unresolvedName && KeyPathNode.INTERNAL_IDS.equals(pathNode.key)) {
+                            return (T) new ArrayList<>(node.internalIds());
```
Comment (tkalkirill) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java:838]: How about an immutable list?

File: modules/configuration/src/test/java/org/apache/ignite/internal/configuration/direct/DirectPropertiesTest.java [hunk 3]
```diff
@@ -246,6 +247,27 @@ public void testNamedListDirectInternalId() throws Exception {
...
+    /**
+     * Checks simple scenarios of getting internalIds of named list.
+     */
+    @Test
+    public void testNamedListDirectInternalIds() throws Exception {
+        DirectConfiguration cfg = registry.getConfiguration(DirectConfiguration.KEY);
+
+        cfg.children()
+                .change(change -> change.create("foo", value -> {
+                }))
```
Comment (tkalkirill) [inline modules/configuration/src/test/java/org/apache/ignite/internal/configuration/direct/DirectPropertiesTest.java:259]:
```suggestion
                .change(change -> change.create("foo", value -> { }))
```

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java (hunk 2 above)
Comment (ibessonov) [reply modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java:838]: Sure

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**